sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java

```diff
@@ -0,0 +1,1086 @@
+            "jsr305.return.without.nullness.annotation";
+
+    /** Key for violation message. */
+    public static final String MSG_OVERRIDDEN_METHODS_ALLOW_ONLY_NONNULL =
+            "jsr305.overridden.methods.allow.only.nonnull";
+
+    /** Key for violation message. */
+    public static final String MSG_NEED_TO_INHERIT_PARAM_ANNOTATIONS =
+            "jsr305.need.to.inherit.param.annotations";
+
+    /** Key for violation message. */
+    public static final String MSG_CONSTRUCTOR_WITH_RETURN_ANNOTATION =
+            "jsr305.constructor.with.return.annotation";
+
+    /** Package name. */
+    private static final String PKG_JAVAX_ANNOTATION = "javax.annotation";
+
+    /**
+     * Class NullnessAnnotation. The annotations we consider as "nullness-relevant".
+     */
+    private enum NullnessAnnotation {
+
+        /** Override. */
+        OVERRIDE("Override", "java.lang"),
+        /** CheckForNull. */
+        CHECK_FOR_NULL("CheckForNull", PKG_JAVAX_ANNOTATION),
+        /** Nullable. */
+        NULLABLE("Nullable", PKG_JAVAX_ANNOTATION),
+        /** Nonnull. */
+        NONNULL("Nonnull", PKG_JAVAX_ANNOTATION),
+        /** CheckReturnValue. */
+        CHECK_RETURN_VALUE("CheckReturnValue", PKG_JAVAX_ANNOTATION),
+        /** ParametersAreNonnullByDefault. */
+        PARAMETERS_ARE_NONNULL_BY_DEFAULT("ParametersAreNonnullByDefault", PKG_JAVAX_ANNOTATION),
+        /** ParametersAreNullableByDefault. */
+        PARAMETERS_ARE_NULLABLE_BY_DEFAULT("ParametersAreNullableByDefault", PKG_JAVAX_ANNOTATION),
+        /** ReturnValuesAreNonnullByDefault. */
+        RETURN_VALUES_ARE_NONNULL_BY_DEFAULT("ReturnValuesAreNonnullByDefault",
+                "edu.umd.cs.findbugs.annotations");
+
+        /** The annotation's name. */
+        private final String annotationName;
+
+        /** The annotation's fully qualified class name. */
+        private final String fullyQualifiedClassName;
+
+        /**
+         * Constructor.
+         * @param annotationName
+         *        the annotation's name
+         * @param packageName
+         *        the package name
+         */
+        NullnessAnnotation(final String annotationName, final String packageName) {
+            this.annotationName = annotationName;
+            this.fullyQualifiedClassName = packageName + "." + annotationName;
+        }
+
+    }
+
+    /** The map of annotations against their respective names. */
+    private static final Map<String, NullnessAnnotation> STRING2ANNOTATION =
+            createString2AnnotationMap();
+
+    /** The modifiers of interest. */
+    private static final int[] ACCEPTABLE_MODIFIERS = {
+        TokenTypes.PARAMETER_DEF,
+        TokenTypes.METHOD_DEF,
+        TokenTypes.PACKAGE_DEF,
+        TokenTypes.CTOR_DEF,
+        TokenTypes.CLASS_DEF,
+        TokenTypes.INTERFACE_DEF,
+        TokenTypes.ENUM_DEF,
+    };
+
+    /** Parameter: packages to check. */
+    private String[] packages = new String[0];
```

Comment:
I'd like to bounce that one back. If you feel that this is a showstopper then I can of course rename it. However since the original CheckStyle plugin has been out and in use for a while this would force people migrating from the old plugin to the check in sevntu to change their configuration which some might find unnecessary.

Reply:
It isn't a showstopper to me, just soemthing I was pointing out since I didn't expect everything to be ignored by default.

Reply:
In that case I propose to add a line to the Javadoc, like: "Note that by default no files will be checked. To enable this check, you need to set the `packages` property to one or more parent packages."

What do you think?

Reply:
Something like this is what I was asking for in the other item, to specifically point out that the user has to configure something before using it.
I am fine with the proposed line.

Reply:
Done.